Added definition of different sorts. (#1119)
TheAlgorithms/C: data_structures/array/README.md

## Sort Function

> The Sort function sorts the elements and arrange in a particular order. The different types of sorting methods are Bubble Sort, Selection Sort,insertion sort, Merge Sort and Quick Sort.

Reviewer: ```suggestion
The Sort function sorts the elements and arranges them in a particular order. The different types of sorting methods are Bubble Sort, Selection Sort, Insertion Sort, Merge Sort and Quick Sort.

```

> 1. Bubble Sort repeatedly sorts the adjacent elements if they are in wrong order.
> 2. Selection sort repeatedly finds minimum elements from unsorted array and place at the beginning, this algorithm maintains two sub array one is sorted and one  is  unsorted.
> 3. Insertion sort makes virtualy two subarrays one is sorted and one is unsorted . Takes values from unsorted array and put at proper position in sorted array.
> 4. Merge sort is based on divide and conquer method. This sortig algorithm continiously divide array in two equal halves and combined in sorted order.
> 5. Quick sort is a divide and conquer algorithm , it takes an element as an pivot and partitions the array around the pivot element.

Reviewer: ```suggestion
1. Bubble Sort repeatedly sorts the adjacent elements if they are in the wrong order.
2. Selection sort repeatedly finds the minimum elements from an unsorted array and places them at the beginning. This algorithm maintains two sub-arrays: one is a sorted array, and the other is unsorted.
3. Insertion sort makes virtually two subarrays: one is a sorted array, and the other is unsorted. Takes values from the unsorted array and puts them in the proper position in a sorted array.
4. Merge sort is based on the divide and conquer method. This sorting algorithm continuously divides the array into two halves and combines them in sorted order.
5. Quick sort is a divide-and-conquer algorithm. It takes an element as a pivot and partitions the array around the pivot element.
```
